packages/runtime-common/realm-auth-data-source.ts

```diff
@@ -6,28 +6,22 @@ export class RealmAuthDataSource {
   private visitedRealms = new Map<string, RealmAuthClient>();
   private matrixClient: MatrixClient;
   private getFetch: () => typeof globalThis.fetch;
-  realmURL: string;
 
   constructor(
     matrixClient: MatrixClient,
     // we want our fetch to be lazily obtained as it might be the very fetch
     // that is composed by middleware containing this data source instance
     getFetch: () => typeof globalThis.fetch,
-    realmURL: string,
   ) {
     this.matrixClient = matrixClient;
     this.getFetch = getFetch;
-    this.realmURL = realmURL;
   }
 
   token(url: string): string | undefined {
     let targetRealmURL = this.toRealmURL(url);
     if (!targetRealmURL) {
       return undefined;
     }
-    if (targetRealmURL === this.realmURL) {
-      return undefined;
-    }
```

Comment:
this was the crucial logic that was blowing away our session token for each HTTP GET. there is no need for this, the worker has no special realm preference. I think this made sense when the worker lived on the realm. but now that workers are their own separate process we need to treat them as arms length from any particular realm